lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+  dt.print_error("incremented file is " .. incremented_filepath)
+
+  return incremented_filepath
+end
+
+--[[
+  NAME
+    dutils.groupIfNotMember - group a new file with an existing one
+
+  SYNOPSIS
+    result = dtutils.groupIfNotMember(image, new_image)
+      image - an existing image in the darktable database
+      new_image - the new image to group with the existing image
+
+  DESCRIPTION
+    groupIfNotMember checks a file to see if it's a member of the group
+    with an existing image.  If not, then the group leader of the existing 
+    image is found and the new image is grouped with it.
+
+  RETURN VALUE
+    none
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.groupIfNotMember(img, new_img)
+  local image_table = img:get_group_members()
+  local is_member = false
+  for _,image in ipairs(image_table) do
+    dt.print_error(image.filename .. " is a member")
+    if image.filename == new_img.filename then
+      is_member = true
+      dt.print_error("Already in group")
+    end
+  end
+  if not is_member then
+    dt.print_error("group leader is "..img.group_leader.filename)
+    new_img:group_with(img.group_leader)
+    dt.print_error("Added to group")
+  end
+end
+
+--[[
+  NAME
+    dutils.sanitize_filename - take care of spaces in filenames
+
+  SYNOPSIS
+    result = dtutils.sanitize_filename(filepath)
+      filepath - an optional path and filename with spaces
+
+  DESCRIPTION
+    sanitize_filename escapes spaces in filenames so that the can be passed as arguments
+
+  RETURN VALUE
+    result - sanitized filename
+
+  ERRORS
+    Only spaces in filenames are sanitized, but not in the path
+
+  
+]]
+
+function dtutils.sanitize_filename(filepath)
+  local path = dtutils.get_path(filepath)
+  local basename = dtutils.get_basename(filepath)
+  local filetype = dtutils.get_filetype(filepath)
+
+  local sanitized = string.gsub(basename, " ", "\\ ")
+
+  return path .. sanitized .. "." .. filetype
+end
+
+--[[
+  NAME
+    dutils.show_status - show the status while exporting images
+
+  SYNOPSIS
+    result = dtutils.show_status(storage, image, format, filename, number, total, high_quality, extra_data)
+      storage - the storage that the status is being shown for
+      image - the current image
+      format - the format of the export image
+      filename - the filename being exported to
+      number - the number of this image in the sequence
+      total - the total number of images being exported
+      high_quality - boolean
+      extra_data - extra data from the storage
+
+  DESCRIPTION
+    show_status runs prior to each image being exported and prints out a status
+
+  RETURN VALUE
+    none
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.show_status(storage, image, format, filename,
+  number, total, high_quality, extra_data)
+    dt.print(string.format(_("Export Image %i/%i"), number, total))
+end
+
+--[[
+  NAME
+    dutils.extract_image_list - assemble the exported image filenames from an image_table into a string
```

Comment:
This is another thing that gets used in exporters a lot, as well as processors.  Maybe a libExporter?